metadata-models/docs/entities/assertion.md

```diff
@@ -0,0 +1,37 @@
+# Assertion
+
+Assertion entity represents a data quality rule applied on dataset.  
+In future, it can evolve to span across Datasets, Flows (Pipelines), Models, Features etc.
+
+## Identity
+
+Assertion is identified by globally unique identifier, which is function of details that uniquely identify the assertion in an assertion platform.
```

Comment:
Consider adding a sentence : For example, we compute the globally unique id for Great Expectations by ... 

Also consider adding a point about how the key is expected to be constant / stable across multiple executions of the same assertion

Reply:
I made this change